azure/application-gateway/variables.tf

```diff
@@ -0,0 +1,207 @@
+  type = list(object({
+    name                           = string
+    frontend_ip_configuration_name = string
+    frontend_port_name             = optional(string, "Http")
+    protocol                       = optional(string, "Http")
+    host_name                      = optional(string)
+    host_names                     = optional(list(string))
+    ssl_certificate_name           = optional(string)
+  }))
+}
+variable "ssl_certificates" {
+  description = "List of SSL Certs"
+  type = list(object({
+    name                = string
+    data                = optional(string)
+    password            = optional(string)
+    key_vault_secret_id = optional(string)
+  }))
+  default = []
+  validation {
+    condition     = try(alltrue([for cert in var.ssl_certificates : (cert.key_vault_secret_id != null || (cert.data != null && cert.password != null))]), true)
+    error_message = "Each certificate must specify either data and password or key_vault_secret_id."
+  }
+}
+variable "url_path_maps" {
+  description = "List of Path Maps"
+  type = list(object({
+    name                               = string
+    default_backend_address_pool_name  = string
+    default_backend_http_settings_name = optional(string, "Default")
+    path_rule = list(object({
+      name                       = string
+      paths                      = list(string)
+      backend_address_pool_name  = string
+      backend_http_settings_name = optional(string, "Default")
+    }))
+  }))
+  default = []
+}
+variable "request_routing_rules" {
+  description = "List of Routing Rules"
+  type = list(object({
+    name                       = string
+    rule_type                  = optional(string, "PathBasedRouting")
+    http_listener_name         = string
+    backend_address_pool_name  = optional(string)
+    backend_http_settings_name = optional(string)
+    url_path_map_name          = optional(string)
+    priority                   = optional(number, 100)
+  }))
+}
+
+variable "identity_ids" {
+  description = "List of potential UserAssigned identities"
+  type        = list(string)
+  default     = null
+}
+
+variable "tags" {
+  description = "Resource Tags"
+  type        = map(string)
+}
+
+variable "probe" {
+  description = "List of Routing Rules"
+  type = list(object({
+    name                                      = optional(string, "Default")
+    protocol                                  = optional(string, "Http")
+    interval                                  = optional(number, 30)
+    path                                      = optional(string, "/")
+    timeout                                   = optional(number, 30)
+    unhealthy_threshold                       = optional(number, 3)
+    port                                      = optional(number, 80)
+    pick_host_name_from_backend_http_settings = optional(bool, false)
+    host                                      = optional(string)
+  }))
+  default = [{
+    name = "Default"
+  }]
+}
```

Comment:
On the settings, do not override null for the port. port` should be left `optional(number)`, inheriting from the backend settings by default